### PR TITLE
Make password and reset_password exclusive

### PR DESCRIPTION
**Pull request by ilpianista**
The documentation works when either `password` or `reset_password` is set.

### CODE REVIEW

**Comment by roidelapluie** [comment]
Can we have an acceptance test with an empty password?

**Comment by ilpianista** [comment]
@roidelapluie is this OK as test? I don't set a password during the creation, but then it's set during the user update test.

Still, I don't think we can check if the password has been set because the password is (obviously) never sent back by GitLab. Thus, I removed that part.

**Comment by ilpianista** [comment]
Making the password optional has been solved by #193, so this PR now just ensure that either `password` or `reset_password` is set, not both.

**Comment by armsnyder** [comment]
@ilpianista Can you add two new tests?
- One where you only set `reset_password`, just to ensure that it doesn't explode if `password` is undefined.
- One where you set neither field and expect an error.

**Comment by ilpianista** [comment]
@armsnyder I implemented 2). To implement 1), I simply removed the `password` argument from the existing test.

File: gitlab/resource_gitlab_user_test.go [hunk 1]
```diff
@@ -96,7 +97,12 @@ func TestAccGitlabUser_password_reset(t *testing.T) {
 		Providers:    testAccProviders,
 		CheckDestroy: testAccCheckGitlabGroupDestroy,
 		Steps: []resource.TestStep{
-			// Create a user
+			// Test that either password or reset_password is needed
+			{
+				Config:      testAccGitlabUserConfigWrong(rInt),
+				ExpectError: regexp.MustCompile(`\sAt least one of either password or reset_password must be defined`),
```
Comment (armsnyder) [inline gitlab/resource_gitlab_user_test.go:103]: Nitpick but the `\s` made me scratch my head for a minute. I think the capital `A` alone serves the intended purpose with less head scratching. Thoughts?

```suggestion
				ExpectError: regexp.MustCompile("At least one of either password or reset_password must be defined"),
```

**Review by armsnyder — APPROVED** [review]
Thanks!

**Pull request merged**